runtime/lua/vim/shared.lua

```diff
@@ -1430,4 +1430,21 @@ function vim._ensure_list(x)
   return { x }
 end
 
+--- Like tonumber(), but for integer values.
+--- @param x any
+--- @return integer?
+function vim._tointeger(x)
+  local nx = tonumber(x)
+  if nx and nx == math.floor(nx) then
```

Comment:
Some edge cases for which this also returns `true`:
- `'inf'` and `'-inf'`. As both `math.huge` and `-math.huge` are equal to their `math.floor()` output.
- Strings with big numbers with decimal parts, like `'100000000000000000000.1'`. Maybe this would be intended, but better be aware of it.

Reply:
Also worth noting, we use `math.huge` in a few places where we expect integer (e.g. in the example of `vim.fs.find()`). However, even though not technically true, we can use `math.huge` in most places we allow an integer.